public/app/features/alerting/unified/reducers/ruler/ruleGroups.ts

```diff
@@ -1,27 +1,35 @@
 import { createAction, createReducer, isAnyOf } from '@reduxjs/toolkit';
-import { remove } from 'lodash';
+import { inRange } from 'lodash';
 
-import { RuleIdentifier } from 'app/types/unified-alerting';
-import { PostableRuleDTO, PostableRulerRuleGroupDTO, RulerRuleDTO } from 'app/types/unified-alerting-dto';
+import { EditableRuleIdentifier, GrafanaRuleIdentifier, RuleIdentifier } from 'app/types/unified-alerting';
+import { PostableRuleDTO, PostableRulerRuleGroupDTO } from 'app/types/unified-alerting-dto';
 
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
 import { hashRulerRule } from '../../utils/rule-id';
-import { isCloudRulerRule, isGrafanaRulerRule } from '../../utils/rules';
+import {
+  isCloudRuleIdentifier,
+  isCloudRulerRule,
+  isGrafanaRuleIdentifier,
+  isGrafanaRulerRule,
+} from '../../utils/rules';
 
 // rule-scoped actions
-export const addRuleAction = createAction<{ rule: PostableRuleDTO }>('ruleGroup/rules/add');
-export const updateRuleAction = createAction<{ identifier: RuleIdentifier; rule: PostableRuleDTO }>(
+export const addRuleAction = createAction<{ rule: PostableRuleDTO; groupName?: string; interval?: string }>(
+  'ruleGroup/rules/add'
+);
+export const updateRuleAction = createAction<{ identifier: EditableRuleIdentifier; rule: PostableRuleDTO }>(
   'ruleGroup/rules/update'
 );
 export const pauseRuleAction = createAction<{ uid: string; pause: boolean }>('ruleGroup/rules/pause');
-export const deleteRuleAction = createAction<{ rule: RulerRuleDTO }>('ruleGroup/rules/delete');
+export const deleteRuleAction = createAction<{ identifier: EditableRuleIdentifier }>('ruleGroup/rules/delete');
 
 // group-scoped actions
 export const updateRuleGroupAction = createAction<{ interval?: string }>('ruleGroup/update');
-export const moveRuleGroupAction = createAction<{ namespaceName: string; groupName?: string; interval?: string }>(
+export const moveRuleGroupAction = createAction<{ newNamespaceName: string; groupName?: string; interval?: string }>(
```

Comment:
Shouldn't `groupName` and `interval` parameters be required?

Reply:
It's perfectly fine to move a rule group to another namespace while keeping the current group name in tact :)